[IMM32] Make IMM32 non-Wine module

katahiromz opened the pull request:
## Purpose

Related to #7870. `IMM32` uses NDK. So, we have to remove Wine dependency.
JIRA issue: [CORE-5743](https://jira.reactos.org/browse/CORE-5743)

## Proposed changes

- Add `dll/win32/imm32/debug.h`.
- Remove Wine dependency.
- Don't use `debugstr_a` and `debugstr_w` macros.
- Update the documentation (`media/doc/WINESYNC.txt`) to reflect the removal of the Wine dependency.

## Testbot runs (Filled in by Devs)

- [x] KVM x86: https://reactos.org/testman/compare.php?ids=101390,101486
- [ ] KVM x64: https://reactos.org/testman/compare.php?ids=101391

katahiromz commented:
BEFORE: 8d7eca9
AFTER: 7704bc4

KVM x86: https://reactos.org/testman/compare.php?ids=101390
KVM x64: https://reactos.org/testman/compare.php?ids=101391

oleg-dubinskiy reviewed (commented):
Looks almost OK to me, but I have two suggestions still:
1. Since imm32 isn't considered as Wine-synced module any more, is it worth to update WINESYNC.txt accordingly (remove the info about imm32)?: https://github.com/reactos/reactos/blob/80eb313eedace26c9453c2eb09f4cf9f85c322eb/media/doc/WINESYNC.txt#L83
2. Maybe we can also get rid from Wine debugging macros too? And use (or just redefine) `DPRINT` and `DPRINT1` instead of `TRACE`, `WARN`, `ERR` and `FIXME`. But this probably needs (much) more code changes, so I think this is an optional proposal.

katahiromz commented:
> 1. Since imm32 isn't considered as Wine-synced module any more, is it worth to update WINESYNC.txt accordingly (remove the info about imm32)?: 

Deleted the line.

> 2. Maybe we can also get rid from Wine debugging macros too? And use (or just redefine) `DPRINT` and `DPRINT1` instead of `TRACE`, `WARN`, `ERR` and `FIXME`. But this probably needs (much) more code changes, so I think this is an optional proposal.

Replaced them with `DPRINT` and `DPRINT1`. Thank you.

katahiromz commented:
BEFORE: 8d7eca9
AFTER: 00411f6

KVM x86: https://reactos.org/testman/compare.php?ids=101390
KVM x64: https://reactos.org/testman/compare.php?ids=101391

katahiromz commented:
BEFORE: 80eb313
AFTER: 089be7b

KVM x86: https://reactos.org/testman/compare.php?ids=101483
KVM x64: https://reactos.org/testman/compare.php?ids=101484

HBelusca commented on dll/win32/imm32/CMakeLists.txt:
```diff
@@ -22,29 +19,12 @@ list(APPEND SOURCE
-    ${CMAKE_CURRENT_SOURCE_DIR}/res/2052_Bitmap_207.bmp
-    ${CMAKE_CURRENT_SOURCE_DIR}/res/2052_Bitmap_208.bmp
-    ${CMAKE_CURRENT_SOURCE_DIR}/res/2052_Bitmap_209.bmp)
+file(GLOB imm32_rc_deps ${CMAKE_CURRENT_SOURCE_DIR}/res/*.*)
```
nice!

katahiromz commented on dll/win32/imm32/candidate.c:
```diff
@@ -7,9 +7,6 @@
 
 #include "precomp.h"
 
-WINE_DEFAULT_DEBUG_CHANNEL(imm);
```
Fixed.

katahiromz commented on dll/win32/imm32/debug.h:
```diff
@@ -0,0 +1,88 @@
+    #define DBG_IS_CHANNEL_ENABLED(ch,level)
+    #define DBG_PRINT(ch,level)
+    #define DPRINT(fmt, ...)
+    #define DPRINT1(fmt, ...)
```
Fixed.

katahiromz commented on dll/win32/imm32/debug.h:
```diff
@@ -0,0 +1,88 @@
+{
+    DbgChimm = 0,
+} DEBUGCHANNEL;
+
```
I will do it later after this PR.
